sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangePartitionKeyImpl.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.feedranges;
+
+import com.azure.cosmos.implementation.Constants;
+import com.azure.cosmos.implementation.IRoutingMapProvider;
+import com.azure.cosmos.implementation.apachecommons.collections.list.UnmodifiableList;
+import com.azure.cosmos.implementation.routing.PartitionKeyInternal;
+import com.azure.cosmos.implementation.routing.Range;
+import com.azure.cosmos.models.PartitionKeyDefinition;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+import static com.azure.cosmos.BridgeInternal.setProperty;
+
+final class FeedRangePartitionKeyImpl extends FeedRangeInternal {
+    private final PartitionKeyInternal partitionKey;
+
+    public FeedRangePartitionKeyImpl(PartitionKeyInternal partitionKey) {
+        if (partitionKey == null) {
+            throw new NullPointerException("partitionKey");
+        }
+
+        this.partitionKey = partitionKey;
+    }
+
+    public PartitionKeyInternal getPartitionKeyInternal() {
+        return this.partitionKey;
+    }
+
+    @Override
+    public void accept(FeedRangeVisitor visitor) {
+        if (visitor == null) {
+            throw new NullPointerException("visitor");
+        }
+
+        visitor.visit(this);
+    }
+
+    @Override
+    public <TInput> void accept(GenericFeedRangeVisitor<TInput> visitor, TInput input) {
```

Comment:
why do we need the both version async and sync accept?

Reply:
Discussed offline - usage of the visitors will be added when consuming the FeedRange Api in feedRanges - will double check then which of them are still needed etc. (I only ported those I believe we will need - but I am porting a moving target - so this is work in progress)